### ISSUE TITLE
NavigateAsync does not navigate with MasterDetail in Prism 7.0.0.269

### ISSUE THREAD

**Issue by Giorgi** [PrismLibrary/Prism#1275]
### Package info

- Platform: Android
- Prism version: 7.0.0.269-pre
- Xamarin version (if applicable): Xamarin.Forms 2.4.0.280

### Repro steps

Please describe the issue/request in detail.

I have a page (MainPage) and a master-detail page (MasterDetail). When the app is launched I show login screen and on login button click I call
`navigationService.NavigateAsync("/MasterDetail/NavigationPage/MainPage");`
but nothing happens.

The same code works perfectly with Prism 6.3.0 and navigates to MasterDetail page with MainPage shown as detail page.

**Comment by brianlagunas** [comment]
Can you provide a sample that reproduces the issue? I am unable to reproduce it locally.

**Comment by brianlagunas** [comment]
Can you wrap you navigate call in a try catch and await it and assign the result to a a variable. Then check if you get an exception, or check the state of the awaited call?

**Comment by Giorgi** [comment]
Found it!

I had a file with build action set to "EmbeddedResource" and looks like when I ported the project from PCL to .netstandard I forgot about it and the file was not being embedded any more. This was causing null reference exception in MasterDetail page ctor and navigation was failing silently. Once I set the correct build action it worked as expected.

By the way I have a handler for `AndroidEnvironment.UnhandledExceptionRaiser` and `AppDomain.CurrentDomain.UnhandledException` but the exception was not being reported there. Perhaps it makes sense to rethrow the exception at least in Debug mode? Or maybe print it to the Output window of VS ?
